addons/account/models/account_move_send.py

```diff
@@ -226,8 +226,7 @@ def _get_placeholder_mail_attachments_data(self, move, invoice_edi_format=None,
         """
         if move.invoice_pdf_report_id:
             return []
-
-        filename = move._get_invoice_report_filename()
+        filename = move._get_invoice_report_filename(report=pdf_report or self._get_default_pdf_report_id(move))
```

Comment:
```suggestion
        filename = move._get_invoice_report_filename(report=pdf_report)
```
You already do the fallback on `_get_invoice_report_filename`